### ISSUE TITLE
Override builtin functions

### ISSUE THREAD

**Issue by t-konstant** [lunet-io/scriban#227]
What would be the best way to override a built-in function? I am trying to override the array.contains function to make it work with [JArray](https://www.newtonsoft.com/json/help/html/T_Newtonsoft_Json_Linq_JArray.htm).
I created a new class which extends ArrayFunctions class trying to override _Contains_ function
```
public class MyArrayFunctions : ArrayFunctions
    {
        public new static bool Contains(IEnumerable list, object item)
        {
           // custom logic
        }
    }
```
and setting the class in my script object like so
```
public class MyScriptObject : BuiltinFunctions
    {
        public MyScriptObject()
        {
            SetValue("array", new MyArrayFunctions(), true);
        }
    }
```
The template context is initialized with _MyScriptObject_

But my custom function is not getting called at all.

Appreciate your help

**Comment by xoofx** [comment]
I'm not sure that the code today honor the "new" and the base class might override everything. Can you double check that renaming your method `Contains`  to e.g `MyContains` would work?

**Comment by t-konstant** [comment]
`MyContains` works. I am using that for now. Thanks!

**Comment by t-konstant** [comment]
`MyContains` works. I will use that in the meantime. Thanks!

**Comment by xoofx** [comment]
This will be fixed in next 2.1.3
